test/fuzz/test_runner.py

```diff
@@ -193,27 +193,54 @@ def main():
         )
 
 
-def generate_corpus(*, fuzz_pool, src_dir, build_dir, corpus_dir, targets):
-    """Generates new corpus.
+def transform_process_message_target(targets, src_dir):
+    """Add a target per process message, and also keep ("process_message", {}) to allow for
+    cross-pollination, or unlimited search"""
+
+    p2p_msg_target = "process_message"
+    has_p2p_msg = (p2p_msg_target, {}) in targets
+    if has_p2p_msg:
+        lines = subprocess.run(
+            ["git", "grep", "--function-context", "g_all_net_message_types{", src_dir / "src" / "protocol.cpp"],
+            check=True,
+            stdout=subprocess.PIPE,
+            text=True,
+        ).stdout.splitlines()
+        lines = [l.split("::", 1)[1].split(",")[0].lower() for l in lines if l.startswith("src/protocol.cpp-    NetMsgType::")]
+        assert len(lines)
+        targets += [(p2p_msg_target, {"LIMIT_TO_MESSAGE_TYPE": m}) for m in lines]
+    return targets
+
+
+def transform_rpc_target(targets, src_dir):
+    """Add a target per RPC command, and also keep ("rpc", {}) to allow for cross-pollination,
+    or unlimited search"""
 
-    Run {targets} without input, and outputs the generated corpus to
-    {corpus_dir}.
-    """
-    logging.info("Generating corpus to {}".format(corpus_dir))
     rpc_target = "rpc"
-    has_rpc = rpc_target in targets
-    if has_rpc:
-        targets.remove(rpc_target)
-    targets = [(t, {}) for t in targets]
+    has_rpc = (rpc_target, {}) in targets
     if has_rpc:
```

Comment:
nit: `has_...` is not really useful anymore i guess
```suggestion
    if (rpc_target, {}) in targets:
```